src/serde/read_cache_lookup.rs

```diff
@@ -185,13 +225,13 @@ impl ReadCacheLookup {
 /// If `A` => `v` then `[A] + [0]` => `v * 2` and `[A] + [1]` => `v * 2 + 1`
 /// Then the integer is turned into the minimal-length array of `u8` representing
 /// that value as an unsigned integer.
-fn reversed_path_to_vec_u8(path: &[u8]) -> Vec<u8> {
+fn reversed_path_to_vec_u8(path: &BitSlice) -> Vec<u8> {
```

Comment:
I'm user switching to bitvec instead of u8 uses less memory, but I don't really know if it would be faster. On the one hand, we get more data in cache; on the other hand, now we're doing a lot of masking. I guess it comes down to if we ever actually blow out the cache, which depends a lot on how big the cpu cache is, which varies a lot.

So I suppose this change is probably never negative.

Reply:
I don't think this made a significant difference either. This was, again, while looking to minimize and avoid heap allocations. I think there's a possibility that this causes fewer allocations, as it might not grow as many times as a `Vec<u8>`